lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -2517,6 +2517,30 @@ def test_dim_aux_coords(self):
             ],
         )
 
+    def test_aux_coords_ordering(self):
+        cube = Cube(np.zeros((3, 4)))
+        cube.add_aux_coord(
+            AuxCoord(np.zeros((3, 4)), long_name="a12", units=1), (0, 1)
+        )
+        cube.add_aux_coord(
+            AuxCoord(np.zeros((3, 4)), long_name="b12", units=1), (0, 1)
+        )
+        cube.add_aux_coord(DimCoord(np.arange(3), long_name="a1", units=1), 0)
+        cube.add_aux_coord(DimCoord(np.arange(3), long_name="b1", units=1), 0)
+        cube.add_aux_coord(DimCoord(np.arange(4), long_name="b2", units=1), 1)
+        cube.add_aux_coord(DimCoord(np.arange(4), long_name="a2", units=1), 1)
+        result = cube._summary_vector_sections_info()
+        self.assertEqual(
+            [[co.name() for co in sect[1]] for sect in result],
+            [
+                [],
+                ["a1", "b1", "a12", "b12", "a2", "b2"],
```

Comment:
This would need changing depending on [my previous comment regarding `vector_aux_coords.sort()`](https://github.com/SciTools/iris/pull/3914/commits/5c9ace49f3080a53bf1eafe691a81c8b96990818#diff-38eaa350bf479c486e9f6659b8077be2d52448f17739872e50285b98b5423415R2352).

Reply:
Turns out this is co-incidentally correct even if we included alphabetical sorting!